### DISCUSSION TITLE
FNIRSI GC-01: Revert to original firmware

### DISCUSSION THREAD

**Discussion by ullix** [Gissio/radpro#125] [Q&A]
Is it possible, to revert a FNIRSI GC-01 device with Radpro back to the original, factory, firmware? How?

**Comment by gc01noob** [comment]
if you download the whole repo and look for the folder called docs, inside there is a GC01 folder then a firmware folder. you'll find the stock .bin for your CPU in there
using any Linux GUI drag the 1.6 version over to the device BOOTLOADER drive window, leave the window open, then drag it over again and overwrite and you'll be back to stock 1.6-2

**Comment by gc01noob** [comment]
ps. you can mess about with Windows if you like but you'll end up pulling your hair out. even Live booting Ubuntu from a USB drive is simpler.

**Comment by ullix** [comment]
@gc01noob Thanks, but all files listed here
https://github.com/Gissio/radpro/tree/main/docs/devices/FNIRSI%20GC-01/firmware
have the comment "...2.0rc3..." or "rc5"? Do I look at the wrong place?

**Reply by dc1rdb** [reply]
This is the right folder with the original firmwares. Disregard the comments.

**Comment by ullix** [comment]
... and each of these files is original firmware? Which one do I pick, then.  My counter has the chip:
`FNIRSI GC-01:   CH32F103 R8T6  401546C10 WCH`

The `*install.txt ` file kind of suggests I could ruin my computer when it fails! Not a very comfortable thought.

**Comment by lsd3k** [comment]
anyone have gc-01 apm32f103rbt6 original firmware v1.8

**Comment by alschirov** [comment]
The same question about 1.8 ?

**Comment by Neodymium5312** [comment]
official firmware link can be found here: https://fnirsistore.com/fnirsi-firmware-upgrade/
latest official original firmware is 1.6 and thats the only link i can currently find, not sure if 1.8 was ever released as a download
